trie/verkle.go

```diff
@@ -163,15 +162,19 @@ func (t *VerkleTrie) UpdateAccount(addr common.Address, acc *types.StateAccount,
 		stem      = t.pointCache.GetTreeKeyBasicDataCached(addr[:])
 	)
 
+	// XXX this is a workaround until we manage to "update subfields"
+	bd, err := t.root.Get(stem, t.FlatdbNodeResolver)
+	if err == nil && len(bd) > 0 {
+		basicData[utils.BasicDataVersionOffset] = bd[utils.BasicDataVersionOffset]
+		copy(basicData[utils.BasicDataCodeSizeOffset:utils.BasicDataNonceOffset], bd[utils.BasicDataCodeSizeOffset:utils.BasicDataNonceOffset])
```

Comment:
this was the issue: it used to copy the entire field, and if for whatever reason the updated value was smaller than the old one, remnants of the old value would remain. This couldn't happen for the nonce, but it could happen with the balance, e.g. if a contract sends its whole balance to another account.

Reply:
Is your comment placed in the right lines?
What I see in this code is adding the version (was missing) and code-size (code was commented) to `basicData`.

Reg the balance, I only see endianess changes, not really not reusing something?

Reply:
before the fix, I was reusing everything, that was the bug.

and yes the comment is in the right place, but it's talking about something that no longer exists.